src/Neo/SmartContract/Native/CryptoLib.cs

```diff
@@ -115,5 +117,27 @@ public static bool VerifyWithECDsaV0(byte[] message, byte[] pubkey, byte[] signa
                 return false;
             }
         }
+
+        /// <summary>
+        /// Verifies that a digital signature is appropriate for the provided key and message using the Ed25519 algorithm.
+        /// </summary>
+        /// <param name="message">The signed message.</param>
+        /// <param name="signature">The signature to be verified.</param>
+        /// <param name="publicKey">The public key to be used.</param>
+        /// <returns><see langword="true"/> if the signature is valid; otherwise, <see langword="false"/>.</returns>
+        [ContractMethod(Hardfork.HF_Echidna, CpuFee = 1 << 15)]
+        public static bool VerifyWithEd25519(byte[] message, byte[] signature, byte[] publicKey)
+        {
+            if (signature.Length != Ed25519.SignatureSize)
+                throw new ArgumentException("Invalid signature size", nameof(signature));
+
+            if (publicKey.Length != Ed25519.PublicKeySize)
+                throw new ArgumentException("Invalid public key size", nameof(publicKey));
```

Comment:
I'm wondering is it OK to throw exception here, because VerifyWithECDsa returns `false` if the format of arguments is invalid:
https://github.com/neo-project/neo/blob/1688a7a3e24ace2372f839d3efd86c22ff6718dc/src/Neo/SmartContract/Native/CryptoLib.cs#L96-L99
So may be we need compatible behaviour (just return `false`), what do you think?

Reply:
I think you are talking about something that should be called `TryVerify`, but fine, i can update anyway.